library-best-practices/JLBP-17.md

```diff
@@ -0,0 +1,61 @@
+[JLBP-17] Coordinate Major Version Adoption
+-------------------------------------------
+
+For any particular library dependency chain, the libraries involved need to
+coordinate the adoption of new major versions of any shared dependency that has
+breaking changes. This is necessary so that consumers can easily use the latest
+version of all of the functionality together.  This best practice doesn't apply
+to the adoption of new major versions where the package was renamed, because the
+old and new packages can co-exist.
+
+Below is an example of what happens when a new major version of a leaf library
+is partially adopted. Suppose gax-java and grpc-java depend on guava 20.0:
+
+```
+gax-java 1.29.0 ->
+  grpc-java 1.13.1 ->
+    guava 20.0
+  guava 20.0
+```
+
+Suppose Guava version 26.0 deletes a public method and it is now the latest
```

Comment:
So this example is suggesting that Guava should coordinate releases with gax-java and grpc-java? That criterion seems only satisfiable in a mono-repo world.

Reply:
Yes, but only for major versions. No coordination needs to be done for minor (= non-breaking) changes.